dali/benchmark/decoder_bench.cc

```diff
@@ -21,352 +21,227 @@
 namespace dali {
 
 class DecoderBench : public DALIBenchmark {
-};
-
-BENCHMARK_DEFINE_F(DecoderBench, HostDecoder)(benchmark::State& st) { // NOLINT
-  int batch_size = st.range(0);
-  int num_thread = st.range(1);
-  DALIImageType img_type = DALI_RGB;
-
-  // Create the pipeline
-  Pipeline pipe(
-      batch_size,
-      num_thread,
-      0, -1,
-      true,   // pipelined
-      2,      // pipe length
-      true);  // async
-
-  TensorList<CPUBackend> data;
-  this->MakeJPEGBatch(&data, batch_size);
-  pipe.AddExternalInput("raw_jpegs");
-  pipe.SetExternalInput("raw_jpegs", data);
-
-  pipe.AddOperator(
-      OpSpec("HostDecoder")
-      .AddArg("device", "cpu")
-      .AddArg("output_type", img_type)
-      .AddInput("raw_jpegs", "cpu")
-      .AddOutput("images", "cpu"));
-
-  // Build and run the pipeline
-  vector<std::pair<string, string>> outputs = {{"images", "cpu"}};
-  pipe.Build(outputs);
-
-  // Run once to allocate the memory
-  DeviceWorkspace ws;
-  pipe.RunCPU();
-  pipe.RunGPU();
-  pipe.Outputs(&ws);
-
-  while (st.KeepRunning()) {
-    if (st.iterations() == 1) {
-      // We will start he processing for the next batch
-      // immediately after issueing work to the gpu to
-      // pipeline the cpu/copy/gpu work
-      pipe.RunCPU();
-      pipe.RunGPU();
-    }
+ public:
+  void DecoderPipelineTest(benchmark::State& st,
```

Comment:
I think it is no longer `DecoderPipelineTest` as adding Decoder is optional now.

Reply:
It's not optional, it's just that the OpSpec is passed as an argument